app/code/Magento/UrlRewrite/Controller/Router.php

```diff
@@ -82,9 +82,9 @@ public function match(RequestInterface $request)
             $this->storeManager->getStore()->getId()
         );
 
-        if ($rewrite === null) {
-            //No rewrite rule matching current URl found, continuing with
-            //processing of this URL.
+        if ($rewrite === null || !$this->isValidRedirect($rewrite)) {
```

Comment:
Will it make PHPMD unhappy if we put condition right here?

Reply:
I think it is normal solution.
Do you mean I should move this logic to another place?